### DISCUSSION TITLE
Decimal as field type

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco.Forms.Issues#1515] [Ideas]
Currently it is possible to add a custom field type storing one of these `FieldDataType`:

- String
- LongString
- Integer
- DateTime
- Bit

![image](https://github.com/umbraco/Umbraco.Forms.Issues/assets/2919859/e8613193-1e96-492c-b20b-c0e363e3bcb0)

However I wonder if there's a specific reason there is no decimal/float/long value?

E.g. compared to value type of label property editor in Umbraco core.

![image](https://github.com/umbraco/Umbraco.Forms.Issues/assets/2919859/7223f3de-4d9b-4ba9-9688-92db43c31dfb)

**Comment by AndyButland** [comment]
No specific reason.  IIRC decimal was added as a value type later in CMS, and it's not been added to Forms.

Could be done.  In terms of prioritising, I'd be interested to to know if find you have use cases for it out in the real world of building projects, or do you just notice it as something missing when you look at the features?

**Comment by bjarnef** [comment]
I don't have a specific usecase at the moment, but for some kind of submission for a ticket and e.g. price (eventually multiple choice for a (half)marathon or event like Codegarden) I would probably use a decimal value for a Price field type. It can of course be done using a string or interger in major units as a workaround.

**Comment by TvGessel** [comment]
This has been added to our backlog for refinement and future pick-up.

**Comment by bjarnef** [comment]
Just came back to this as have some complex forms, where we store store calculations in a custom field type and since the value may contain decimals, we can't use `Integer`.

We can of course use `String` and parse afterwards, but isn't ideal.

Another use case is when working with payments, where one may want to specific an amount. Either this need to be specified in minor units, as string or it can't includes decimals.
